### DISCUSSION TITLE
ParallelLimiter on base class

### DISCUSSION THREAD

**Discussion by fvaillancourt** [thomhurst/TUnit#820] [Q&A] [answered]
I'm considering migrating our XUnit tests to TUnit. Many of our tests are integration tests and require some setup at the beginning to work. So not to repeat the code for each integration test class, we have a base class that does all the setup and teardown and actual test classes inherit from it. Also as they are integration tests, we cannot really run a ton of them at the same time so I applied ParallelLimiter attribute on the base class. However when on the base class it is ignore, it only works if I put it on each test class that derive from the base class. Other attributes like Before works from the base class but not the ParallelLimiter. Is it by design? Is there a good reason or is it a bug? Code example:
```
[ParallelLimiter<Limit1>] // This is ignored ...
public class IntegrationTest
{
    [Before(Test)]
    public void Setup()
    {
        // Called Before
    }
}

// But would work if put here
public class SomeTest : IntegrationTest
{
    [Test]
    public async Task Test1()
    {
    }

    [Test]
    public async Task Test2()
    {
    }
}

public class Limit1 : IParallelLimit
{
    public int Limit => 1;
}
```

**Comment by thomhurst** [comment]
It should apply to subclasses too so an oversight on my part I think. I'll take a look!

**Reply by thomhurst** [reply, marked as answer]
@fvaillancourt Does this work for you on the latest version now?

**Reply by fvaillancourt** [reply]
It does thank you. Also works for `TestExecutor` (probably all, which is good)
